### PR TITLE
PWX-36878: DMthin OCP VSphere pool expand test update

### PR DESCRIPTION
**Pull request by pp511**
Autopilot:
- PvcAndPoolExpand test updated to use resizedisk

storage_pool_test.go:
- triggerPoolExpansion updated to accound for dmthin disk add not supported behavior

pool_function_test.go:
* All tests that were a part of the DMthin test suite were updated to account for the add disk behavior change for dmthin
* PoolExpandSmoky renamed to PoolExpandSmoke
* PoolExpandRejectConcurrent renamed to PoolExpandRejectConcurrentDiskAdd and added PoolExpandRejectConcurrentDiskResize
* PoolExpandWithReboot renamed to PoolExpandDiskAddWithReboot and added PoolExpandDiskResizeWithReboot and 
* PoolExpandWithPXRestart  renamed to PoolExpandDiskAddPXRestart and added PoolExpandDiskResizePXRestart
* Added PoolExpandInvalidSize
* PoolExpandDiskAddAndVerifyFromOtherNode renamed to PoolExpandDiskAddAndVerifyFromOtherNode and added PoolExpandDiskResizeAndVerifyFromOtherNode
* Updated testrail IDs for few tests. More shall be updated after https://purestorage.atlassian.net/browse/PWX-36688

### CODE REVIEW

File: tests/basic/pool_function_test.go [hunk 1]
```diff
@@ -174,10 +163,94 @@ var _ = Describe("{PoolExpandSmoky}", func() {
...
+	It("Expand a pool while a previous expansion is in progress", func() {
+		expandType := api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK
+		targetSize := poolToResize.TotalSize/units.GiB + 100
+		err = Inst().V.ExpandPool(poolIDToResize, expandType, targetSize, true)
+		isExpandInProgress, expandErr := poolResizeIsInProgress(poolToResize)
+		if expandErr != nil {
+			log.Fatalf("Error checking if pool expansion is in progress: %v", expandErr)
+		}
+		if !isExpandInProgress {
+			log.Warnf("Pool expansion already finished. Skipping this test. Using a testing app that writes " +
```
Comment (lsrinivas-pure) [inline tests/basic/pool_function_test.go:242]: I think it should be resize-disk and also I think it should be "Use a testing app that writes"

File: drivers/scheduler/k8s/specs/aut-postgres/aut-postgres-storage.yaml [hunk 2]
```diff
@@ -45,4 +45,14 @@ spec:
...
   resources:
     requests:
-      storage: 1Gi
\ No newline at end of file
+      storage: 1Gi
+---
+##### Required for OpenShift 4.14 or newer
+apiVersion: v1
+kind: ConfigMap
+metadata:
```
[the remark this answers is not in the document]
Comment (pp511) [reply drivers/scheduler/k8s/specs/aut-postgres/aut-postgres-storage.yaml]: Done

File: tests/basic/pool_function_test.go (hunk 1 above)
Comment (pp511) [reply tests/basic/pool_function_test.go]: Done.

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Review by vinayakshnd — APPROVED** [review]
LGTM

**Pull request merged**